niimpy/survey.py

```diff
@@ -76,29 +76,63 @@
     'nearly-every-day': 3
 }
 
-def convert_to_numerical_answer(df, answer_col='raw_answer', prefix=['PSS', 'PHQ2', 'GAD2']):
-    '''
-    Convert text answers into numerical value (assuming a long dataframe).
-    '''
+ID_MAP_PREFIX = {'PSS' : PSS_ANSWER_MAP,
+                 'PHQ2' : PHQ2_ANSWER_MAP,
+                 'GAD2' : PHQ2_ANSWER_MAP}
+
+ID_MAP = {'PHQ2_1':'PSS10_MAP',
+          'PHQ2_2':'PSS10_MAP',
+          'PSQI_1':'PSS10_MAP',
+          'PSQI_2':'PSS10_MAP',
+          'PSQI_3':'PSS10_MAP',
+          'PSQI_4':'PSS10_MAP',
+          'PSQI_5':'PSS10_MAP',
+          'PSQI_6':'PSS10_MAP',
+          'PSQI_7':'PSS10_MAP',
+          'PSQI_8':'PSS10_MAP',
+          'PSQI_9':'PSS10_MAP'}
+
+def convert_to_numerical_answer(df, answer_col, encoded_column, question_id, id_map, use_prefix=False):
```

Comment:
encoded_column is not used, so it can be removed from the params list

Reply:
nice